lib/configure.js

```diff
@@ -142,16 +142,33 @@ function configure (gyp, argv, callback) {
     variables.standalone_static_library = gyp.opts.thin ? 0 : 1
 
     if (vsSetup) {
-      // GYP doesn't (yet) have support for VS2017, so we force it to VS2015
+      // GYP doesn't (yet) have support for VS2017/VS2019, so we force it to VS2015
       // to avoid pulling a floating patch that has not landed upstream.
       // Ref: https://chromium-review.googlesource.com/#/c/433540/
       gyp.opts.msvs_version = '2015'
       process.env['GYP_MSVS_VERSION'] = 2015
       process.env['GYP_MSVS_OVERRIDE_PATH'] = vsSetup.path
       defaults['msbuild_toolset'] = 'v141'
       defaults['msvs_windows_target_platform_version'] = vsSetup.sdk
-      variables['msbuild_path'] = path.join(vsSetup.path, 'MSBuild', '15.0',
-                                            'Bin', 'MSBuild.exe')
+
+      var msbuildPath;
+      try {
+        // Visual Studio 2017
+        msbuildPath = path.join(vsSetup.path, 'MSBuild', '15.0', 'Bin', 'MSBuild.exe')
+        if (!fs.existsSync(msbuildPath)) {
```

Comment:
How about inversing the logic here and prioritize rather VS2019 (Current) directory than 15.0. IMHO current approach would prevent users from compiling vs2019 projects when they have installed both 2017 and 2019 version of VS because msbuild 15.0 would be discovered in the first place. If it would be inverted, MSBuild 2019 would correctly detect older build tools as well. What do you think guys?

Reply:
I agree and fix by https://github.com/nodejs/node-gyp/pull/1715/commits/29b0c2e381f824fa4e641d8659274f5616703d87 . Thanks for good suggestion.